spec/02-integration/05-proxy/11-error_default_type_spec.lua

```diff
@@ -2,32 +2,32 @@ local helpers = require "spec.helpers"
 local cjson   = require "cjson"
 
 
-local S502_MESSAGE = "An invalid response was received from the upstream server"
-local RELOAD_DELAY = 1.0
+local RESPONSE_CODE    = 504
+local RESPONSE_MESSAGE = "The upstream server is timing out"
 
 
-describe("error_default_type", function()
+describe("Proxy errors Content-Type", function()
   local client
 
   setup(function()
     helpers.dao:truncate_tables()
 
     assert(helpers.dao.apis:insert {
-      name         = "api-1",
-      upstream_url = helpers.mock_upstream_url .. "/status/500",
-      hosts        = {
-        "example.com",
-      },
+      name                     = "api-1",
+      methods                  = "GET",
+      upstream_url             = "http://httpbin.org:81",
```

Comment:
I agree that wanting to eliminate flakiness in specs is a worthy goal, but I would really want this done without adding back httpbin into our test suite - even if, as in this tests, it's only to elicit a timeout.

Ideally I would like this to happen with no dependencies - using localhost or the mock upstream. But I understand that might not be achievable in dev machines easily (OSX will not allow the timeout, it will close the socket right away, unless a global setting is changed).

If we're going to use an external url, I suggest we use one that we control, like `http://konghq.com:81` or `http://getkong.com:81`. That way if the httpbin guys decide to change their server config or take down the name our tests will not be affected.

Reply:
@kikito Good point. Let me reassure you - this isn't the intended (final) changeset to be proposed. Several builds failed using the local interface (https://travis-ci.org/Kong/kong/builds), and before leaving yesterday night, I decided to switch to what our [upstream timeout tests](https://github.com/Kong/kong/blob/master/spec/02-integration/05-proxy/06-upstream_timeouts_spec.lua#L36) already do. We can switch it to one of our organization's domains.